library/psa_crypto_slot_management.h

```diff
@@ -113,13 +137,20 @@ void psa_wipe_all_key_slots(void);
  * If multi-threading is enabled, the caller must hold the
  * global key slot mutex.
  *
- * \param[out] volatile_key_id   On success, volatile key identifier
- *                               associated to the returned slot.
+ * \param[out] volatile_key_id   If null, reserve a cache slot for
```

Comment:
I would rephrase this. IMO this parameter only makes sense when the function call returns successfully, however "on success" is only mentioned later. I think we can either assume "on success" implicitly or state that at the beginning for both volatille and built-in keys.

Question: why can't we return a key id also for persistent/built-in keys the same way as we would do for volatile?

Reply:
For a persistent/built-in key, the key ID is chosen by the application, so the caller knows it already. Only the slot management code can figure out the id→slot mapping, because it has found which slot contains the cache entry (maybe because it has just triggered the loading).